### PR TITLE
Document immutable roles

### PR DESCRIPTION
**Pull request by renetapopova**
Replaces #1984

### CODE REVIEW

File: modules/ROOT/pages/authentication-authorization/manage-roles.adoc [hunk 1]
```diff
@@ -573,9 +579,19 @@ This is equivalent to running `DROP ROLE myrole IF EXISTS` followed by `CREATE R
...
-* The `CREATE OR REPLACE ROLE` command does not allow you to use the `IF NOT EXISTS`.
+The `CREATE OR REPLACE ROLE` command does not allow you to use the `IF NOT EXISTS`.
 ====
 
+[role=new-in-5.26]
+[[access-control-immutable-roles]]
+== Immutable roles
+
+Immutable roles are those that cannot be modified in the usual way.
+This means they cannot be created, renamed, dropped, or have privileges granted to or revoked from them.
```
Comment (renetapopova) [inline modules/ROOT/pages/authentication-authorization/manage-roles.adoc:590]: This contradicts the above commands, for example, CREATE [IMMUTABLE] ROLE name AS COPY OF otherName or CREATE OR REPLACE [IMMUTABLE] ROLE name [AS COPY OF otherName], etc.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

File: modules/ROOT/pages/authentication-authorization/immutable-roles-privileges.adoc [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
++
+[source, cypher, role=noplay]
+----
+GRANT IMMUTABLE MATCH {*} ON GRAPH * ELEMENTS * TO analyst
+----
+
+. Ensure you have completed steps 4 and 5 from <<administer-immutable-roles-and-privileges>>.
++
+Now, even users with `ROLE MANAGEMENT` and `PRIVILEGE MANAGEMENT` privileges will not be able do any of the following:
+[source, cypher, role=noplay]
```
Comment (renetapopova) [inline modules/ROOT/pages/authentication-authorization/immutable-roles-privileges.adoc:112]:
```suggestion
+
.Drop the `analyst` role
[source, cypher, role=noplay]
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

File: modules/ROOT/pages/authentication-authorization/immutable-roles-privileges.adoc [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+|"PUBLIC"
+|true
+
+7+a|Rows: 2
+|===
++
+The result shows that all users are restricted from adding or removing privileges, including the `admin` user.
+. Ensure you have completed steps 4 and 5 from <<administer-immutable-roles-and-privileges>>.
+
+=== Creating permanent roles which cannot be changed
```
Comment (renetapopova) [inline modules/ROOT/pages/authentication-authorization/immutable-roles-privileges.adoc:87]:
```suggestion
=== Creating permanent roles that cannot be changed
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

File: modules/ROOT/pages/authentication-authorization/immutable-roles-privileges.adoc [hunk 4]
```diff
@@ -0,0 +1,140 @@
+[role=enterprise-edition not-on-aura]
+[[immutable-roles-and-privileges]]
+= Immutable roles and privileges
+:description: This section explains how to use Cypher to manage immutable roles and privileges.
+
+[NOTE]
+====
+Immutable roles are available starting with Neo4j 5.26.
+====
```
Comment (renetapopova) [inline modules/ROOT/pages/authentication-authorization/immutable-roles-privileges.adoc:9]:
```suggestion
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-operations-2001.surge.sh

New pages:
- [Immutable roles and privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/immutable-roles-privileges)

Updated pages:
- [Database privileges](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/database-administration)
- [Role-based access control](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-privileges)
- [Manage roles](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/authentication-authorization/manage-roles)
- [The Neo4j Operations Manual v5](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5)
- [Tutorials and examples](https://neo4j-docs-operations-2001.surge.sh/operations-manual/5/tutorial)

File: modules/ROOT/pages/authentication-authorization/immutable-roles-privileges.adoc [hunk 5]
```diff
@@ -0,0 +1,136 @@
+[role=enterprise-edition not-on-aura]
+[[immutable-roles-and-privileges]]
+= Immutable roles and privileges
+:description: This section explains how to use Cypher to manage immutable roles and privileges.
+
+
+Immutable privileges are useful for restricting the actions of users who can themselves administer xref:authentication-authorization/dbms-administration.adoc#access-control-dbms-administration-privilege-management[privileges]. 
+Immutable roles are useful for providing _system roles_, which appear as permanent parts of the DBMS.
```
Comment (renetapopova) [inline modules/ROOT/pages/authentication-authorization/immutable-roles-privileges.adoc:8]:
```suggestion
Starting with Neo4j 5.26, Neo4j also introduces immutable roles.
Immutable roles are useful for providing _system roles_, which appear as permanent parts of the DBMS.
```

File: modules/ROOT/pages/authentication-authorization/immutable-roles-privileges.adoc [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+----
+DROP ROLE analyst
+----
++
+.Revoke the `MATCH` privilege from the `analyst` role
+[source, cypher, role=noplay]
+----
+REVOKE MATCH {*} ON GRAPH * ELEMENTS * FROM analyst
+----
+.Rename the `analyst` role
```
Comment (renetapopova) [inline modules/ROOT/pages/authentication-authorization/immutable-roles-privileges.adoc:121]:
```suggestion
+
.Rename the `analyst` role
```

**Pull request merged**